srtcore/sync.cpp

```diff
@@ -189,6 +189,66 @@ std::string srt::sync::FormatTimeSys(const steady_clock::time_point& timestamp)
     return out.str();
 }
 
+srt::sync::Mutex::Mutex()
+{
+    pthread_mutex_init(&m_mutex, NULL);
+}
+
+srt::sync::Mutex::~Mutex()
+{
+    pthread_mutex_destroy(&m_mutex);
+}
+
+int srt::sync::Mutex::lock()
+{
+    return pthread_mutex_lock(&m_mutex);
+}
+
+int srt::sync::Mutex::unlock()
+{
+    return pthread_mutex_unlock(&m_mutex);
+}
+
+bool srt::sync::Mutex::try_lock()
+{
+    return (pthread_mutex_lock(&m_mutex) == 0);
```

Comment:
This should probably be using `pthread_mutex_trylock()`

Reply:
Fixed